lib/mempool/mempool.js

```diff
@@ -209,18 +219,27 @@ class Mempool extends EventEmitter {
       const entry = this.getEntry(hash);
 
       if (!entry) {
+        // Confirmed TX was not in mempool
+        // maybe clear other references and conflicts
         this.removeOrphan(hash);
         this.removeDoubleSpends(tx);
```

Comment:
Related to `removeDoubleSpends`:

There needs to be additional check when the transaction that we have not seen is removed by `removeDoubleSpends`. I think `evictEntry` is a good place to clean up `disconnects`, because evicting the original transaction that we are waiting to reconnect will leave them there w/o any way to clean up.

Same as `removeDoubleSpends` it could happen on covenant invalidation, e.g. unconfirmed disconnector transaction, that we are waiting for got invalidated in the mempool (e.g. expire). This also leads to evictEntry.

I believe all evictEntry callers are included here.

Reply:
Another thing that we need to track is double spends to the .children transactions. Any of the `children` can get double spent in the block, but since they are not part of the normal transaction map and are not tracked via `getSpent` we will need to track them separately.